app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java

```diff
@@ -179,35 +176,48 @@ public static Disposable checkNewVersion(@NonNull final App app) {
 
         // Check if user has enabled/disabled update checking
         // and if the current apk is a github one or not.
-        if (!prefs.getBoolean(app.getString(R.string.update_app_key), true) || !isGithubApk(app)) {
+        if (!prefs
+            .getBoolean(app.getString(R.string.update_app_key), true) || !isGithubApk(app)) {
+            return null;
+        }
+
+        final String expire = prefs.getString(app.getString(R.string.update_expire_date_key), null);
```

Comment:
You should add more comments to your code. NewPipe is pretty bad at this. Please check out https://standards.mousepawmedia.com/csi.html.

Here, you could like above state that you check whether a timeout has been met already.

It does seem a bit wasteful to always re-parse the string. But I presume it's just easier to do that, so you can conveniently use `isAfter`.

Reply:
> It does seem a bit wasteful to always re-parse the string. But I presume it's just easier to do that, so you can conveniently use isAfter

When i added a log statement to the new version check it is only run at startup of the app and not on rotation, meaning very rarely. Is the tradeof of using a little bit more ram constantly compared to a little bit of cpu once good?

@B0pol How did you come to think thats it's run on device rotation

Reply:
As said, I don't mind this. In fact, I probably reconstructed your thinking.

It was an example of what kind of comment is missing there. I can see _what_ you are doing, but only guess _why_ you are doing it like that. It showcases perfectly why CSI matters.

Reply:
@xiangronglin I was wrong about device rotation, it is the case for MainFragment, MainActivity and many other classes .onCreate() methods, but not App.onCreate()

Reply:
I prefer to write the code in a way that does not require comments. In that case maybe add a seperate method with a name, which makes the comment not needed.

I did not do it here at first, since the issue was labelled as `ASAP`. But i forgot that i have time until at least saturday when the release needs to be done

Reply:
> I prefer to write the code in a way that does not require comments. In that case maybe add a seperate method with a name, which makes the comment not needed.

This is just not possible. And very much _not_ a matter of preference.

Do I really have to quote CSI? You don't seem to have had a look:

> “Self-Commenting Code” is a practice wherein a code’s functionality is self-evident. Through naming, structure, and various other techniques, the immediate purpose becomes obvious to the reader. This is beneficial in any language.
> 
> However, “Self-Commenting Code” is seldom capable of expressing the entire intent, or “why”, of the code.
> 
> It is nearly impossible to express the intended behavior of the code;
> 
> -- https://standards.mousepawmedia.com/csi.html#csi-vs-self-commenting-code

Again, your code tells _nothing_ about _why_ you're doing it like that. It cost me valuable time guess the _why_, and that's then still just a guess.

> In practice, intent-commenting saves me considerably more time than it takes! It's a true investment in the future: an extra 10-20 minutes now saves me literal hours later.
> 
> -- https://dev.to/codemouse92/to-comment-or-not-to-comment-3f7h